library/src/main/assets/leaflet_map.html

```diff
@@ -263,32 +263,40 @@
   map.fitBounds(bounds);
 }
 
-function paddingLatLngFromLatLng(latLng) {
-  var centerPoint = map.latLngToContainerPoint(latLng);
-  centerPoint.x = centerPoint.x - (paddingLeft - paddingRight) / 2;
-  centerPoint.y = centerPoint.y - (paddingTop - paddingBottom) / 2;
-  return map.containerPointToLatLng(centerPoint);
+function paddingLatLngFromLatLng(latlng, zoom) {
+  var point = map.project(latlng, zoom);
+  point.x = point.x - (paddingLeft - paddingRight) / 2
+  point.y = point.y - (paddingTop - paddingBottom) / 2;
+  return map.unproject(point, zoom);
 }
 
-function latLngFromPaddingLatLng(paddingLatLng) {
-  var centerPoint = map.latLngToContainerPoint(paddingLatLng);
-  centerPoint.x = centerPoint.x + (paddingLeft - paddingRight) / 2;
-  centerPoint.y = centerPoint.y + (paddingTop - paddingBottom) / 2;
-  return map.containerPointToLatLng(centerPoint);
+function latLngFromPaddingLatLng(latlng, zoom) {
+  var point = map.project(latlng, zoom);
+  point.x = point.x + (paddingLeft - paddingRight) / 2
+  point.y = point.y + (paddingTop - paddingBottom) / 2;
+  return map.unproject(point, zoom);
 }
 
 function centerMap(lat, lng) {
+  centerZoomMap(lat, lng, map.getZoom())
+}
+
+function centerZoomMap(lat, lng, zoom) {
   var latLng = convertLatLng([lat, lng]);
-  map.panTo(paddingLatLngFromLatLng(L.latLng(latLng[0], latLng[1])));
+  map.setView(paddingLatLngFromLatLng(L.latLng(latLng[0], latLng[1]), zoom), zoom, {animate:false});
```

Comment:
Use setView to disable animation and support zoom, same behavior with native-google-map